PluginCode/src/main/java/smr/cs/ualberta/libcomp/DatabaseAccess.java

```diff
@@ -481,16 +484,67 @@ public void updateUserProfile(User userRecord) throws IOException {
 
         if (Integer.parseInt(userRecord.getCloudStore()) == 1) {
             sendUser(userRecord.getUserID(), obj.toString());
+
+            sendFeedbackToCloud(); // check if the user has local feedback stored previously, send them now.
         }
     }
 
-    public void updateFeedback(ReplacementFeedback replacementFeedback) throws IOException {
+    public void storeLocalFeedback(JSONObject obj) throws IOException {
+
+        String filePath = this.filePath +"\\Feedback.json";
+        File myFile = new File(filePath);
+        JSONObject Mainobj = null; // = new JSONObject();
+        if (myFile.exists()) {
+            String content = FileUtils.readFileToString(myFile, "utf-8");
+            JSONObject objM = new JSONObject(content);
+            JSONArray jsonarr = objM.getJSONArray("Feedbacks");
+            jsonarr.put(obj);
+            Mainobj = new JSONObject();
+            Mainobj.put("Feedbacks", jsonarr);
+        } else
+        {
+            Mainobj = new JSONObject();
+            JSONArray array = new JSONArray();
+            array.put(obj);
+            Mainobj.put("Feedbacks", array);
+        }
+
+        FileOutputStream fOuts = new FileOutputStream(myFile);
+        String result = Mainobj.toString();
+        OutputStreamWriter myOutWriter = new OutputStreamWriter(fOuts);
+        myOutWriter.append(result);
+        myOutWriter.close();
+    }
+
+    public void sendFeedbackToCloud() throws IOException {
+        String filePath = this.filePath +"\\Feedback.json";
+        String tokenValue = getUserToken();
+        File myFile = new File(filePath);
+        boolean successfulsent = false;
+        if (myFile.exists()) {
+            String content = FileUtils.readFileToString(myFile, "utf-8");
+            JSONObject objM = new JSONObject(content);
+            JSONArray jsonarr = objM.getJSONArray("Feedbacks");
+            int i = 0;
+            while (i<jsonarr.length() )
+            {
+                JSONObject jsonObj = (JSONObject)jsonarr.get(i);
+                jsonObj.put("user", this.userid);
+                String JsonString = jsonObj.toString();
+                if (sendFeedback(JsonString, tokenValue)) {successfulsent = true;}
+                i++;
+            }
+            if (successfulsent) {Files.deleteIfExists(Paths.get(filePath));}
+
+        }
+    }
+
+    public void updateFeedback(boolean cloudSelect, ReplacementFeedback replacementFeedback) throws IOException {
 
         String tokenValue = getUserToken();
         JSONObject obj = new JSONObject();
         DateFormat df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
-
-        obj.put("user", this.userid);
+        if (cloudSelect) {obj.put("user", this.userid);} // no userid to be stored locally, as it is not yet set
```

Comment:
what does cloudSelect do?

Reply:
renamed to sendToCloud, which is a boolean for whether or not it should be sent or stored only locally